src/LoggingFormats.jl

```diff
@@ -74,39 +66,76 @@ maybe_stringify_exceptions((e, bt)::Tuple{Exception,Any}) = sprint(Base.display_
 maybe_stringify_exceptions(e::Exception) = sprint(showerror, e)
 maybe_stringify_exceptions(v) = v
 
-function JSONLogMessage{T}(args) where {T}
-    JSONLogMessage{T}(
-        lvlstr(args.level),
-        args.message isa AbstractString ? args.message : string(args.message),
-        args._module === nothing ? nothing : string(args._module),
-        args.file,
-        args.line,
-        args.group === nothing ? nothing : string(args.group),
-        args.id === nothing ? nothing : string(args.id),
-        Dict{String,T}(string(k) => transform(T, maybe_stringify_exceptions(v)) for (k, v) in args.kwargs)
+unclash_key(k) = k in STANDARD_KEYS ? Symbol("_", k) : k
+
+function to_namedtuple(::Type{T}, args; nest_kwargs) where {T}
+    kw = (k => transform(T, maybe_stringify_exceptions(v)) for (k, v) in args.kwargs)
+    if nest_kwargs
+        kw = (:kwargs => Dict{String, T}(string(k) => transform(T, maybe_stringify_exceptions(v)) for (k, v) in args.kwargs),)
+    else
+        kw = (unclash_key(k) => transform(T, maybe_stringify_exceptions(v)) for (k, v) in args.kwargs)
+    end
+    return (;
+        level=lvlstr(args.level),
+        msg=args.message isa AbstractString ? args.message : string(args.message),
+        :module => args._module === nothing ? nothing : string(args._module),
+        file=args.file,
+        line=args.line,
+        group=args.group === nothing ? nothing : string(args.group),
+        id=args.id === nothing ? nothing : string(args.id),
+        kw...
```

Comment:
it would be nice to tie the `STANDARD_KEYS` explicitly to the keys of the log message named tuple. something like this maybe

```suggestion
    standard_message = NamedTuple{STANDARD_KEYS}(
        lvlstr(args.level),
        args.message isa AbstractString ? args.message : string(args.message),
        args._module === nothing ? nothing : string(args._module),
        args.file,
        args.line,
        args.group === nothing ? nothing : string(args.group),
        args.id === nothing ? nothing : string(args.id),
    )
    return merge(standard_message, kw)
```